src/index.js

```diff
@@ -68,6 +74,65 @@ class DelegatedPeerRouting {
       log('findPeer finished: ' + id)
     }
   }
+
+  /**
+   * Attempt to find the closest peers on the network to the given key
+   * @param {Uint8Array} key A CID like key
+   * @param {object} options
+   * @param {number} options.timeout How long the query can take. Defaults to 30 seconds
+   * @returns {AsyncIterable<{ id: PeerId, multiaddrs: Multiaddr[] }>}
+   */
+  async * getClosestPeers (key, options = {}) {
+    key = new CID(key)
+    const keyStr = key.toString()
+
+    log('getClosestPeers starts:', keyStr)
+    options.timeout = options.timeout || DEFAULT_TIMEOUT
+
+    const onStart = defer()
+    const onFinish = defer()
+
+    this._httpQueue.add(() => {
+      onStart.resolve()
+      return onFinish.promise
+    })
+
+    try {
+      await onStart.promise
+
+      const peers = new Map()
+
+      for await (const result of this.dht.getClosestPeers(keyStr, {
+        timeout: options.timeout
+      })) {
+        switch (result.type) {
+          case 1: // Found Closer
+            // Track the addresses, so we can yield them when done
+            result.responses.forEach(response => {
+              peers.set(response.id, {
+                id: PeerId.createFromCID(response.id),
+                multiaddrs: response.addrs
+              })
+            })
+            break
+          case 2: // Final Peer
+            yield peers.get(result.id.string) || {
+              id: PeerId.createFromCID(result.id),
+              multiaddrs: []
+            }
+            break
+          default:
+            log('getClosestPeers unhandled response', result)
+        }
+      }
+    } catch (err) {
+      log.error('getClosestPeers errored:', err)
+      throw err
+    } finally {
+      onFinish.resolve()
+      log('getClosestPeers finished:', keyStr)
```

Comment:
perhaps track if we got the peer or not?
otherwise the log might be false positive

Reply:
If there are no peers returned it should throw and the error will be logged in the catch. This should only ever happen if the delegate has no peers in its routing table, which shouldn't happen.